### PR TITLE
Criu ns tty fix

### PR DESCRIPTION
**Pull request by Snorch**
These are (probable) fixes for two errors when using criu-ns:

1. When restoring first time via criu-ns (from criu-ns dump):

```
Error (criu/tty.c:689): tty: Failed to set group 40816 on 0: Inappropriate ioctl for device
```

2. When dumping second time via criu-ns:

```
(00.005678) Error (criu/files-reg.c:1710): Can't lookup mount=29 for fd=0 path=/dev/pts/20
```

Fixes: #1893

### CODE REVIEW

**Comment by Snorch** [comment]
I'm not too sure about second patch, maybe we should skip inherited fds somewhere else, I'm open to any advise.

**Comment by avagin** [comment]
lgtm

File: scripts/criu-ns [hunk 1]
```diff
@@ -124,6 +126,7 @@ def wrap_restore():
     criu_pid = os.fork()
     if criu_pid == 0:
         os.setsid()
+        fcntl.ioctl(0, termios.TIOCSCTTY, 1)
```
Comment (avagin) [inline scripts/criu-ns:129]: pls write a comment that explains what we are doing here and why do we need to do this

Comment (Snorch) [reply scripts/criu-ns]: Done, also I updated the patch logic a bit to skip this step when not needed (not shell job) and fail on notatty.

File: scripts/criu-ns [hunk 2]
```diff
@@ -124,6 +126,18 @@ def wrap_restore():
     criu_pid = os.fork()
     if criu_pid == 0:
         os.setsid()
+        '''
+        Set stdin tty to be a controlling tty of our new session, this is
+        required by --shell-job option, as for it CRIU would try to set a
+        process group of restored root task to be a foreground group on the
+        terminal.
+        '''
```
Comment (rst0git) [inline scripts/criu-ns:134]: The [convention](https://peps.python.org/pep-0257/) in Python is to always use `"""` (triple double quotes) for docstrings.
For [block comments](https://peps.python.org/pep-0008/#block-comments), each line of starts with a `#` and a single space.

Comment (Snorch) [reply scripts/criu-ns:134]: Thanks, will fix!

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**